scenario_player/runner.py

```diff
@@ -2,42 +2,172 @@
-from scenario_player.services.utils.interface import ServiceInterface
-from scenario_player.utils import TimeOutHTTPAdapter, get_udc_and_token, wait_for_txs
-from scenario_player.utils.token import Token, UserDepositContract
+from scenario_player.node_support import NodeController, RaidenReleaseKeeper
+from scenario_player.utils import TimeOutHTTPAdapter, wait_for_txs
+from scenario_player.utils.configuration.settings import SettingsConfig, UDCSettingsConfig
+from scenario_player.utils.token import (
+    load_token_configuration_from_file,
+    token_maybe_mint,
+    token_registry_maybe_register_token,
+    userdeposit_maybe_deposit,
+    userdeposit_maybe_increase_allowance,
+)
 
 if TYPE_CHECKING:
     from scenario_player.tasks.base import Task, TaskState
 
 log = structlog.get_logger(__name__)
 
 
+def get_udc_and_corresponding_token(
+    udc_settings: UDCSettingsConfig, settings: SettingsConfig, proxy_manager: ProxyManager
+) -> Tuple[UserDeposit, Token]:
+    """ Return contract proxies for the UserDepositContract and associated token """
+    assert udc_settings.enable
+
+    udc_address = udc_settings.address
+
+    if udc_address is None:
+        chain_id = settings.chain_id
+        assert chain_id, "Missing configuration, either set udc_address or the chain_id"
+
+        contracts = get_contracts_deployment_info(chain_id, version=RAIDEN_CONTRACT_VERSION)
+
+        msg = (
+            f"invalid chain_id, {chain_id} is not available for version {RAIDEN_CONTRACT_VERSION}"
+        )
+        assert contracts, msg
+
+        udc_address = contracts["contracts"][CONTRACT_USER_DEPOSIT]["address"]
+
+    user_deposit_proxy = proxy_manager.user_deposit(
+        UserDepositAddress(to_bytes(udc_address)), "latest"
+    )
+
+    token_address = user_deposit_proxy.token_address("latest")
+    token_proxy = proxy_manager.token(token_address, "latest")
+
+    return user_deposit_proxy, token_proxy
+
+
+def determine_run_number(scenario_dir: Path) -> int:
+    """ Determine the current run number.
+
+    We check for a run number file, and use any number that is logged there
+    after incrementing it.
+    """
+    # TODO: Use advisory file locks to avoid concurrent read/writes on the
+    # run_number_file. Otherwise this can lead to very hard to debug races.
+    run_number_file = scenario_dir.joinpath(RUN_NUMBER_FILENAME)
+
+    if run_number_file.exists():
+        run_number = int(run_number_file.read_text()) + 1
+    else:
+        run_number = 0
+
+    run_number_file.write_text(str(run_number))
+
+    return run_number
+
+
+def make_session(auth: str, settings: SettingsConfig) -> Session:
+    session = Session()
+    if auth:
+        session.auth = cast(Tuple[str, str], tuple(auth.split(":")))
+    session.mount("http", TimeOutHTTPAdapter(timeout=settings.timeout))
+    session.mount("https", TimeOutHTTPAdapter(timeout=settings.timeout))
+
+    return session
+
+
+def wait_for_token_network_discovery(
```

Comment:
This corresponds to the [`ScenarioRunner.wait_for_token_network_discovery`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/runner.py#L180-L226) which again doesn't need the instance state.